MLResurrection.cpp

```diff
@@ -0,0 +1,668 @@
+			    datatools::service_manager& flServices,
+			    dpp::module_handle_dict_type& /*moduleDict*/){
+
+  // Extract the filename_out key from the supplied config, if the key exists.
+  // datatools::properties throws an exception if the key isn't in the config, so catch this if thrown
+  try {
+    myConfig.fetch("filename_out", filename_output);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Problem in the output file " << e.what() << std::endl;
+  };
+
+  try {
+    myConfig.fetch("dead_cells", dead_cells_list);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Error in configuration files: file with list of dead cells not present " << e.what() << std::endl;
+  };
+
+  try {
+    myConfig.fetch("method", method);
+  }
+  catch (std::logic_error& e) {
+    std::cerr << "Error in configuration files: method not defined " << e.what() << std::endl;
+  };
+
+  std::string string_filename = std::string("reco_hits_")+method+std::string(".txt");
+  const char *reco_hits_filename = string_filename.c_str();
+  std::rename("reco_hits.txt", reco_hits_filename);
+  reco_hits_file << "ID SIDE LAYER ROW" <<std::endl;
+
+  std::cout << "MLResurrection module initialized..." << std::endl;
+  std::cout << "Method selected: " << method << std::endl;
+  std::cout << "Output file: " << filename_output << std::endl;
+
+  flSimParameters = FLSimulate::FLSimulateArgs::makeDefault();
+
+  datatools::service_manager services("DeadCells", "SuperNEMO Dead Cells");
+  datatools::properties services_config;
+  services_config.store("name", "DeadCells");
+  services_config.store("description","SuperNEMO Demonstrator Dead Cells module");
+  std::vector<std::string> scf = {"@falaise:snemo/demonstrator/geometry/" "GeometryService.conf"};
+  services_config.store("services.configuration_files", scf);
+  services.initialize(services_config);
+
+  // Output metadata management:
+  datatools::multi_properties flSimMetadata("name", "type", "Metadata associated to a flsimulate run");
+  do_metadata(flSimParameters, flSimMetadata);
+
+  // Metadata management:
+  simOutput.set_name("DeadCellsOutput");
+  simOutput.set_single_output_file(filename_output);
+  datatools::multi_properties& metadataStore =  simOutput.grab_metadata_store();
+  metadataStore = flSimMetadata;
+  simOutput.initialize_simple();
+
+  // set up CD data bank
+  new_calibrated_data = &(workItem.add<snemo::datamodel::calibrated_data>("CD"));
+
+  // Read the dead cells from a file into the vector:
+  std::cout << "Reading dead cells from " << dead_cells_list << "..." << std::endl;
+  dead_cells_service(dead_cells);
+
+  // print list of dead cells
+  //for (Long64_t i=0; i<N_dead_cells; i++)
+  //  std::cout << dead_cells[i][0] << "-" << dead_cells[i][1] << "-" << dead_cells[i][2] << std::endl;
+
+  this->_set_initialized(true);
+
+}
+
+//! [MLResurrectionModule::Process]
+dpp::base_module::process_status MLResurrectionModule::process(datatools::things& event) {
```

Comment:
Please use consistent indent levels, as well as braces to bound loops/conditionals, even one line ones. It's rather difficult at the moment to work out what's going on at each level as written.